src/Mailgun/Model/Suppression/Unsubscribe/Unsubscribe.php

```diff
@@ -15,49 +15,101 @@
 class Unsubscribe
 {
     /**
+     * The unsubscribe event identifier
+     * 
      * @var string
      */
-    private $address;
+    private $id = NULL;
+    
+    /**
+     * The unsubscribe email address
+     * @var string
+     */
+    private $address = NULL;
 
     /**
+     * Tag to unsubscribe from, use * to unsubscribe address from domain
+     * 
+     * Note: If NULL the value means * (all for a domain)
+     * 
      * @var string
      */
-    private $tag;
+    private $tag = NULL;
 
     /**
+     * The creation date for the event
+     * 
      * @var \DateTime
      */
-    private $createdAt;
+    private $createdAt = NULL;
 
     /**
      * @param string $address
      */
-    private function __construct($address)
+    private function __construct($address = NULL)
     {
         $this->address = $address;
         $this->createdAt = new \DateTime();
     }
 
     /**
-     * @param array $data
+     * Create a Unsubscribe object based in info provided in $data array
+     * 
+     * Fields allowed:
+     * 
+     * - id: The unsubscribe event identifier
+     * - address: The unsubscribe email address
+     * - tag: Tag to unsubscribe (use * or NULL to unsubscribe address from domain)
+     * - created_at: The creation date for the event
+     * 
+     * @param array $data the info provided
      *
      * @return Unsubscribe
      */
     public static function create(array $data)
     {
         $unsubscribe = new self($data['address']);
 
+        if (isset($data['id'])) {
+            $unsubscribe->setId($data['id']);
+        }
+        
         if (isset($data['tag'])) {
             $unsubscribe->setTag($data['tag']);
         }
+        
         if (isset($data['created_at'])) {
             $unsubscribe->setCreatedAt(new \DateTime($data['created_at']));
         }
 
         return $unsubscribe;
     }
 
+	/**
+	 * Get the event identifier
+	 * 
+     * @return string The event identifier
+     */
+    public function getId()
+    {
+        return $this->id;
+    }
+    
+    /**
+     * Set the event identifier 
+     * 
+     * @param string $id The event identifier
+     */
+    private function setId($id = NULL)
```

Comment:
Why should this parameter be optional?

Reply:
It is a way to guarantee at least NULL if not provided or received by the rest client or so. Also since was not previously loaded or used avoid future errors with people code. The really nice thing is if I could put private function setId(string $id = NULL) : Unsubscribe  (using the PHP 7.1 syntax), but it seems that 5.5 and 5.6 are still supported.

Reply:
IMHO if:

```php
if (isset($data['id'])) {
    $unsubscribe->setId($data['id']);
}
```

Then there's no need to allow this

Reply:
The id attribute is on the server spec, but not always is set, so why is not needed?

Reply:
What I mean is:
- If we don't provide a value for a given property, ten by default the value is `null`
- If we don't have the `id` key on the array, then there's no need to call the setter
- If the setter is just setting the value then there's no need to have the setter - due we should not change IDs of external resources, in this case some data that comes from an API - so you can set the value in this way: `$unsubscribe->id = $data['id']`